compiler/lib/code.ml

```diff
@@ -387,9 +387,9 @@ module Constant = struct
           done;
           !same
     | Int64 a, Int64 b -> Some (Int64.equal a b)
-    | Float_array a, Float_array b -> Some (Array.equal Poly.equal a b)
+    | Float_array a, Float_array b -> Some (Array.equal Float.equal a b)
     | Int a, Int b -> Some (Int32.equal a b)
-    | Float a, Float b -> Some (Poly.equal a b)
+    | Float a, Float b -> Some (Float.equal a b)
```

Comment:
I think we need to keep `Poly.equal`. In Ocaml, `nan = nan` is false, but `Float.equal nan nan` is true.